### ISSUE TITLE
Updating feathr package to Python >3.8

### ISSUE THREAD

**Issue by xiaoyongzhu** [linkedin/feathr#401]
https://endoflife.date/python

Python 3.7 will no longer be supported in a year by the community so we might need to move Feathr to 3.8 only in a year

**Comment by blrchen** [comment]
Agree, since both databricks and synapse defaults to use python 3.8, we should make 3.8 as default and minamal?
